features/step_definitions/restrict-pr.js

```diff
@@ -0,0 +1,159 @@
+
+        this.pipelines[rootDir] = {
+            pipelineId: this.pipelineId
+        };
+    }
+);
+
+When(
+    /^a branch is created "([^"]*)" organization$/,
+    {
+        timeout: TIMEOUT
+    },
+    async function step(sourceOrg) {
+        const postfix = new Date().getTime().toString();
+        const sourceBranch = `test-branch-${postfix}`;
+
+        this.sourceOrg = sourceOrg;
+        this.sourceBranch = sourceBranch;
+
+        await github
+            .createBranch(this.sourceBranch, this.sourceOrg, this.repoName)
+            .catch(() => Assert.fail('Failed to create branch.'));
+    }
+);
+
+When(
+    /^a new file is added to the "([^"]*)" directory$/,
+    {
+        timeout: TIMEOUT
+    },
+    async function step(rootDir) {
+        await github
+            .createFile(this.sourceBranch, this.sourceOrg, this.repoName, rootDir)
+            .catch(() => Assert.fail('Failed to create file.'));
+    }
+);
+
+When(
+    /^a pull request is opened from the "([^"]*)" organization$/,
+    {
+        timeout: TIMEOUT
+    },
+    async function step(sourceOrg) {
+        await github
+            .createPullRequest(`${sourceOrg}:${this.sourceBranch}`, this.targetBranch, this.repoOrg, this.repoName)
+            .then(({ data }) => {
+                this.pullRequestNumber = data.number;
+                this.sha = data.head.sha;
+            })
+            .catch(() => Assert.fail('Failed to create the Pull Request.'));
+    }
+);
+
+Then(
+    /^the PR job of "([^"]*)" is triggered because it is not restricted$/,
+    {
+        timeout: TIMEOUT
+    },
+    async function step(rootDir) {
+        const build = await sdapi.searchForBuild({
+            instance: this.instance,
+            pipelineId: this.pipelines[rootDir].pipelineId,
+            desiredSha: this.sha,
+            desiredStatus: ['QUEUED', 'RUNNING', 'SUCCESS', 'FAILURE'],
+            jobName: this.jobName,
+            pullRequestNumber: this.pullRequestNumber,
+            jwt: this.jwt
+        });
+
+        const jobs = await sdapi.findJobs({
+            instance: this.instance,
+            pipelineId: this.pipelines[rootDir].pipelineId,
+            jwt: this.jwt
+        });
+
+        const jobData = jobs.body;
+        const prJobName = `PR-${this.pullRequestNumber}:${this.jobName}`;
+        const job = jobData.find(j => j.name === prJobName);
+
+        Assert.equal(build.jobId, job.id);
+    }
+);
+
+Then(
+    /^the PR job of "([^"]*)" is not triggered because it is restricted$/,
+    {
+        timeout: TIMEOUT
+    },
+    async function step(rootDir) {
+        const build = await sdapi.findBuilds({
+            instance: this.instance,
+            pipelineId: this.pipelines[rootDir].pipelineId,
+            jobName: this.jobName,
+            pullRequestNumber: this.pullRequestNumber,
+            jwt: this.jwt
+        });
+
+        let result = build.body || [];
+
+        result = result.filter(item => item.sha === this.sha);
```

Comment:
This `Then` statement is used [by twice in a scenario](https://github.com/screwdriver-cd/screwdriver/blob/e3849dd4d473fc791945a2d9c1f72eba2e4fb62b/features/restrict-pr.feature#L34-L35), but `this.sha` seems to contains only one sha, that commit is for `all` directory(last commit). Can it test for `branch` directory ?

Reply:
thx.
I fixed it.

Reply:
Couldn't you fix the scenario to contains multiple sha for each commits instead of splitting scenario?
I concerns how long time it takes to execute for too many scenario.

Reply:
The code has been modified to work in the scenario before the modification.
The fix is to use `rootDir` and `jobName` instead of `sha` for unique identification.